drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -2600,8 +2600,8 @@ void RasterizerSceneGLES3::render_scene(const Ref<RenderSceneBuffers> &p_render_
 			spec_constant_base_flags |= SceneShaderGLES3::DISABLE_FOG;
 		}
 
-		if (render_data.environment.is_null() || (render_data.environment.is_valid() && environment_get_fog_mode(render_data.environment) == RS::EnvironmentFogMode::ENV_FOG_MODE_DEPTH)) {
-			spec_constant_base_flags |= 1 << SPEC_CONSTANT_USE_DEPTH_FOG;
+		if (render_data.environment.is_valid() && environment_get_fog_mode(render_data.environment) == RS::EnvironmentFogMode::ENV_FOG_MODE_DEPTH) {
+			spec_constant_base_flags |= 1 << SceneShaderGLES3::USE_DEPTH_FOG;
```

Comment:
Whoops, this should just be an `|=`
```suggestion
			spec_constant_base_flags |= SceneShaderGLES3::USE_DEPTH_FOG;
```